pycbc/workflow/jobsetup.py

```diff
@@ -275,21 +275,20 @@ def multi_ifo_coherent_job_setup(workflow, out_files, curr_exe_job,
         tags = []
     data_seg, job_valid_seg = curr_exe_job.get_valid_times()
     curr_out_files = FileList([])
-    if 'IPN' in datafind_outs[-1].description \
-            and 'bank_veto_bank' in datafind_outs[-2].description:
-        # FIXME: This looks like a really nasty hack for the GRB code.
-        #        This should be fixed properly to avoid strange behaviour!
-        ipn_sky_points = datafind_outs[-1]
-        bank_veto = datafind_outs[-2]
-        frame_files = datafind_outs[:-2]
-    else:
-        ipn_sky_points = None
-        if 'bank_veto_bank' in datafind_outs[-1].name:
-            bank_veto = datafind_outs[-1]
-            frame_files = datafind_outs[:-1]
-        else:
-            bank_veto = None
-            frame_files = datafind_outs
+    ipn_sky_points = None
+    veto_file = None
+    bank_veto = None
+    frame_files = FileList(datafind_outs)
+    for f in datafind_outs:
+        if 'IPN' in f.description:
+            ipn_sky_points = f
+            frame_files.remove(f)
+        elif 'vetoes' in f.description:
+            veto_file = f
+            frame_files.remove(f)
+        elif 'INPUT_BANK_VETO_BANK' in f.description:
+            bank_veto = f
+            frame_files.remove(f)
```

Comment:
A couple comments on this block:
1. It looks like `datafind_outs` and `frame_files` may contain items that do not correspond to frame files. The variable names should then be more general, e.g. `file_list`.
2. The string "IPN" is just three characters long. How high is the risk of clashing with random occurrences of the same string in other file names? Does it make sense to use "_IPN_"?
3. In the spirit of potentially helping debug issues arising from point 2, please add logging statements at each decision, e.g. "Using %s for the IPN sky points", "Using %s for the veto file".

Reply:
Thanks, @titodalcanton.  I renamed `frame_files` `input_files`.  The search is now for `IPN_SKY_POINTS` which is more robust and consistent with the naming that is set up [here](https://github.com/gwastro/pycbc/blob/master/bin/pygrb/pycbc_make_offline_grb_workflow#L272).
Do you think the logging is still necessary?  I'm asking because with `IPN_SKY_POINTS` it more robust and jobsetup.py has not `logging` usage so this may be an intended choice (@spxiwh?).

Reply:
I definitely would not expect clashes if you use IPN_SKY_POINTS, so indeed you can skip point 3.